### ISSUE TITLE
Feature Request: Add option to have LLM refactor suggestion instead of just y/n

### ISSUE THREAD

**Issue by kennethcassel** [AbanteAI/archive-old-cli-mentat#2]
<img width="463" alt="image" src="https://github.com/biobootloader/mentat/assets/22961671/0cd0fa9e-4030-4203-b355-102dcaad7647">

Would be cool if when presented with multiple changes in interactive mode, you could give feedback for updates to a specific function during that flow. 

i.e. 

in this example if I wanted mentat to make the function more concise or to use a different method for calculating primes, I could just ask it to update it's suggestion.

**Comment by biobootloader** [comment]
Good idea! You can already give feedback on an entire set of changes (just type anything instead of y/n/i), but once in interactive mode it would be useful to give feedback to a specific change.

For the UI, two options I see:
- give feedback on that specific example, then move on to approving/rejecting/giving feedback on the other changes. Then, mentat would create updated versions of each change you provided feedback on
- when you give feedback, it rewrites the specific example right there, repeating until you accept or reject it, and then you move on to the next change in the original list.

Complications:
- do we give the entire context again just to rewrite one change? most of it is probably not necessary, and users would have to pay for those tokens.
- what happens if the feedback implies the need for a separate other change as well (e.g. if you ask it to rewrite to use some function that it now has to import at the top of the file)

**Comment by gileneusz** [comment]
some ideas of other options:

Chat with the change: It could be beneficial to implement an option for the user to have a more in-depth conversation with the AI about the specific change. This would involve taking the entire context into consideration, guiding the AI towards generating a different modification.

Try again: Another option could be to essentially "reset" the change and re-prompt the AI. This would effectively erase the previous AI response from the context and allow for a fresh approach to the modification.

Minor modification: Yet another possibility would be to enable the user to make minor adjustments to the AI-generated response. I do believe, however, that in this case it would still be essential to keep the entire context intact in order to yield the most accurate results.

Critique: This could be another interesting approach where the user critiques the AI's response, considering different scenarios or alternate paths (somewhat implementing a 'chain of thoughts'). This could generate multiple alternative responses, from which the user selects the one they find most appropriate. This chosen answer then stays in the conversation context while the other alternatives are removed, leading to a more nuanced and user-controlled outcome.

**Comment by inventor2525** [comment]
Why not just handle things in a branch with full merge request kind of behavior?

Append to the bottom of the context any changes that the user makes in the same diff format as the model, only assign a role, and allow for a comment section. If the user wants then to comment on a change without changing it themself, the model can see where they're talking about.

Also, I really like context modification! Would be especially nice to be able to pause generation and change the name or even the entire flow and just let it complete from there.
